providers/cscglobal/cscglobalProvider.go

```diff
@@ -0,0 +1,68 @@
+package cscglobal
+
+import (
+	"fmt"
+	"sort"
+	"strings"
+
+	"github.com/StackExchange/dnscontrol/v3/models"
+	"github.com/StackExchange/dnscontrol/v3/providers"
+)
+
+/*
+
+CSC Global Registrar:
+
+Info required in `creds.json`:
+   - api-key             Api Key
+   - user-token          User Token
+   - notification_emails (optional) Comma seperated list of email addresses to send notifications to
+*/
+
+func init() {
+	providers.RegisterRegistrarType("CSCGLOBAL", newCscGlobal)
+}
+
+func newCscGlobal(m map[string]string) (providers.Registrar, error) {
+	api := &api{}
+
+	api.key, api.token = m["api-key"], m["user-token"]
+	if api.key == "" || api.token == "" {
+		return nil, fmt.Errorf("missing CSC Global api-key and/or user-token")
+	}
+
+	if m["notification_emails"] != "" {
+		api.notifyEmails = strings.Split(m["notification_emails"], ",")
+	}
+
+	return api, nil
+}
+
+// GetRegistrarCorrections gathers corrections that would being n to match dc.
+func (c *api) GetRegistrarCorrections(dc *models.DomainConfig) ([]*models.Correction, error) {
+	nss, err := c.getNameservers(dc.Name)
+	if err != nil {
+		return nil, err
+	}
+	foundNameservers := strings.Join(nss, ",")
+
+	expected := []string{}
+	for _, ns := range dc.Nameservers {
+		name := strings.TrimRight(ns.Name, ".")
```

Comment:
I'm going to bet that the API is consistent about including or not including the trailing dot.

TrimRight will remove multiple "."; it would be odd if "domain.tld.." was the string.  It could be a configuration error. In that case, we don't want to obscure the error by correcting the user's mistake (TrimRight). Instead we should propagate the mistake so that it is visible and will be corrected.

Suggestion:

```
if ns.Name[len(ns.Name)-1] !=  '.' {
    return nil, fmt.Errorf("Should not happen.  Maybe the API changed?  Please check.")
    // When this code was written ns.Name always includes a single trailing dot.
    // If that changes, the code should change too.
}
name := ns.Name[0:len(ns.Name)-2]
```

Reply:
Correct - turns out it never includes it the trailing dot.